python/tvm/script/scope_handler.py

```diff
@@ -282,6 +282,12 @@ def block(axes=None, name_hint: str = "", span: Optional[Span] = None):
                 if block_info.writes
                 else []
             )
+
+            region_detect_mask: int = (block_info.reads is None) | ((block_info.reads is None) << 1)
```

Comment:
should be block_info.writes?

please add an regression test for write detection